cli/azd/pkg/project/framework_service_dotnet.go

```diff
@@ -191,3 +210,39 @@ func normalizeDotNetSecret(key string) string {
 	// ":".
 	return strings.ReplaceAll(key, "__", ":")
 }
+
+// findProjectFile locates the project file to pass to the `dotnet` tool for a given dotnet service. In the case where
+// [Path] for the the service results in a directory, if current directory contains a single file matching the
+// glob `*.*proj`, then project file will be returned. If there are multiple or no project file, error is returned
+// and user will be asked to add project file path under "project" in azure.yaml.
+func findProjectFile(serviceConfig *ServiceConfig) (string, error) {
```

Comment:
Generally, you want util functions to take minimal input, and if you think about the logic here -- it's almost irrelevant to `azd`. So I would recommend accepting `serviceName string, projectPath string` as input args, instead of `serviceConfig` completely.

Reply:
Not sure if it helps, but it also simplifies the documentation for the function to now read:

`projectPath is either a path to a directory, or to a project file. When projectPath is a directory, the first file matching the glob expression *.*proj (what dotnet expects) is returned. If multiple files match, an error is returned`